dockstore-webservice/src/main/java/io/dockstore/webservice/resources/UsernameRenameRequiredFilter.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2021 OICR and UCSC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.dockstore.webservice.resources;
+
+import io.dockstore.webservice.CustomWebApplicationException;
+import io.dockstore.webservice.core.User;
+import java.security.Principal;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.container.ResourceInfo;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.ext.Provider;
+import org.apache.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@UsernameRenameRequired
+@Provider
+public class UsernameRenameRequiredFilter implements ContainerRequestFilter {
+
+    private static final Logger LOG = LoggerFactory.getLogger(UsernameRenameRequired.class);
+
+    @Context
+    private ResourceInfo resourceInfo;
+
+    @Override
+    public void filter(ContainerRequestContext requestContext) {
+        if (requestContext != null) {
+            final Principal principal = requestContext.getSecurityContext().getUserPrincipal();
+            if (principal instanceof User) {
+                final User user = (User)principal;
+                if (user.isUsernameChangeRequired()) {
+                    throw new CustomWebApplicationException("Your username contains one or more of the following keywords: dockstore, admin, curator, system, or manager. "
```

Comment:
May not be worth it, but it would be nice if these words and the regex had the same source. If we decided to ban "ucsc" one day, we need to update both this message and the regex.

Reply:
Moved both to the Constants file